### ISSUE TITLE
Potential SC2016 false positive when ending single quotes and starting them again

### ISSUE THREAD

**Issue by tdmalone** [koalaman/shellcheck#1160]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2016
- My shellcheck version (`shellcheck --version` or "online"): 0.4.7
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/bash

curl "https://hooks.slack.com/services/${SLACK_WEBHOOK}" -X POST -H 'Content-Type: application/json' --data '{"text":"Directory `'"${DIRECTORY}"'` has been successfully deployed to S3 bucket `'"${BUCKET}"'` from `'"$( hostname )"'`.","username":"Static Site Deployer","icon_emoji":":female-construction-worker:"}'

```

#### Here's what shellcheck currently says:

^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.
(pointing at the single quote after `"{$DIRECTORY}"`)

^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.
(pointing at the single quote after `"{$BUCKET}"`)

#### Here's what I wanted or expected to see:

Nothing!

**Comment by tdmalone** [comment]
So, my intention in this case was to _end_ the single quotes, start double quotes for an expansion, then end the double quotes and start single quotes again to continue my string. I've done that twice (actually, three times including the subshell), and Shellcheck thinks I am still using single quotes.

I may be using my quotes wrong... happy to be corrected!

**Comment by Nightfirecat** [comment]
I believe this is one of those cases where you're using backticks within quotes correctly, but this could, in other cases, be construed as an improper expansion. You'll want to [disable SC2016 for that line](/koalaman/shellcheck/wiki/disable).

**Comment by koalaman** [comment]
Yes, ShellCheck just noticed you have `` ` `` in a single quoted string, and in this case that was intentional.

It's definitely a false positive, but I don't see a systematic way of avoiding it. Please ignore it with a directive.
